### PR TITLE
+ includes with nested items

### PR DESCRIPTION
**Pull request by pape-io**
Enables including nested items inside arrays:
```
# place
{
  category_relations: [
    { href: "http://datastore/category/2" }
  ]
}
```

```ruby
Place.includes(:category_relations)
# fetches now nested items as well
```

### CODE REVIEW

File: lib/lhs/concerns/record/request.rb [hunk 1]
```diff
@@ -46,6 +46,10 @@ def extend_raw_data(data, addition, key)
             link = item[key.to_sym]
             link.merge_raw!(addition[i]) if link.present?
           end
+        elsif data[key]._raw.is_a? Array
+          data[key].each_with_index do |item, index|
+            item._raw.merge!(addition[index]._raw)
+          end
```
Comment (Averethel) [inline lib/lhs/concerns/record/request.rb]: how about:
```ruby
data[key].zip(additions).each do |item, addition|
  item._raw.merge(addition._raw)
```

**Comment by Averethel** [comment]
small comment otherwise 👍

File: lib/lhs/concerns/record/request.rb (hunk 1 above)
Comment (pape-io) [inline lib/lhs/concerns/record/request.rb]: I have to modify the originals??? so how does this help @Averethel ?

Comment (pape-io) [inline lib/lhs/concerns/record/request.rb]: missing a `!`

Comment (Averethel) [inline lib/lhs/concerns/record/request.rb]:
```ruby
a = {}
b = {b: 1}
[a].zip([b]).each{ |a, b| a.merge!(b) }

a # =>  {b: 1}
```

**Pull request merged**